### PR TITLE
fix: Allow NONE licenses without logging errors (also completes comments for license types)

### PR DESCRIPTION
**Pull request by sreyemnayr**
### What I did
Added keys for intentionally unlicensed (not to be confused with the "Unlicense" 🙃 ) source code without an error.

fixes: #

### Checklist

- [x] Passes all linting checks (pre-commit and CI jobs)
- [x] New test cases have been added and are passing
- [x] Documentation has been updated
- [x] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Review by fubuloubu — DISMISSED** [review]
Really like these updates, but curious what others think

**Comment by antazoey** [comment]
Yes, this lgtm, the UI offers the same so it makes sense.

File: ape_etherscan/verify.py
Comment (mikeshultz) [inline ape_etherscan/verify.py:60]: issue: This is not the same thing.  "UNLICENSED" is not the same as The Unlicense.  The SPDX identifier being `Unlicensed` for the former and `Unlicense` for the latter.  `Unlicensed` is actually technically not valid and not listed in the SPDX list.

In fact, this confusion [is not new, which we can see from this Stack Exchange discussion](https://opensource.stackexchange.com/questions/12410/is-there-a-difference-between-unlicense-org-and-spdx-unlicensed).

[The Solidity docs](https://docs.soliditylang.org/en/latest/layout-of-source-files.html#spdx-license-identifier) are more inline to what I thought `UNLICENSED` meant.  They state:

> If you do not want to specify a license or if the source code is not open-source, please use the special value UNLICENSED. Note that UNLICENSED (no usage allowed, not present in SPDX license list) is different from UNLICENSE (grants all rights to everyone).

**Review by mikeshultz — CHANGES_REQUESTED** [review]

File: ape_etherscan/verify.py
Comment (mikeshultz) [reply ape_etherscan/verify.py:60]: Here's [Etherscan's license reference](https://etherscan.io/contract-license-types) to add to the confusion.  Code `2` does in fact mean "The Unlicense".  If we're making these description modifiers here, I think it might be a good idea to update this var name to `UNLICENSE` or something to reduce confusion a bit.

Comment (sreyemnayr) [reply ape_etherscan/verify.py:60]: Yeah, I went back and forth and decided to split the difference… return 1 for “UNLICENSED” in the spdx identifier but use “NO_LICENSE” for the enum key as it’s more in line with the etherscan docs (since this is the Etherscan plugin)

Comment (sreyemnayr) [reply ape_etherscan/verify.py:60]: Ah, yeah the key for 2 should be “UNLICENSE” not “UNLICENSED”

Comment (sreyemnayr) [reply ape_etherscan/verify.py:60]: Oh I see what you mean, the key for 2 should be `UNLICENSE` not `UNLICENSED`

Comment (mikeshultz) [reply ape_etherscan/verify.py:60]: Ah, looks like you added `unlicensed` key is correct in `_SPDX_ID_TO_API_CODE` and it maps the the correct Etherscan ID, I missed that on my first pass.  :+1: 

I guess just the Enum key should be `UNLICENSE` rather than `UNLICENSED` to reduce confusion.

**Comment by mikeshultz** [comment]
@sreyemnayr is there anything you need to get this across the line?

**Comment by sreyemnayr** [comment]
Just time and a nudge 😄 will try to get it tomorrow!

**Comment by fubuloubu** [comment]
ah crap @antazoey, looks like plugins need the treatment of https://github.com/ApeWorX/ape/pull/2020 as well

**Comment by antazoey** [comment]
> ah crap @antazoey, looks like plugins need the treatment of [ApeWorX/ape#2020](https://github.com/ApeWorX/ape/pull/2020) as well

Yes, they do, everything does! It's horrendous. The issue has been open with Github for so long.

**Comment by antazoey** [comment]
> ah crap @antazoey, looks like plugins need the treatment of [ApeWorX/ape#2020](https://github.com/ApeWorX/ape/pull/2020) as well

here is this one: https://github.com/ApeWorX/ape-etherscan/pull/126
I am not going to fix all 300 of our python packages; we will just fix them as they come up.

**Comment by antazoey** [comment]
tests failed, fixed in core: https://github.com/ApeWorX/ape/pull/2022

**Comment by sreyemnayr** [comment]
Is there something else that needs doing now that the tests are fixed?

File: ape_etherscan/verify.py
Comment (antazoey) [inline ape_etherscan/verify.py:58]: Our standard for markdown is 1 sentence per line

**Review by antazoey — APPROVED** [review]

**Comment by fubuloubu** [comment]
> @antazoey looks like there's still some test issues here?  We need to set an infura key for this repo?

New contributors don't have access to our secrets, so just need to check it out of band

**Review by mikeshultz — APPROVED** [review]
Approving assuming the tests are passing.  :shrug:

**Comment by mikeshultz** [comment]
Confirmed tests pass, but I can't merge this because passing tests are required.  @fubuloubu you want to merge this?

**Pull request merged**
